clusterloader2/pkg/test/simpletestexecutor.go

```diff
@@ -28,20 +33,72 @@ func createSimpleTestExecutor() TestExecutor {
 
 // ExecuteTest executes test based on provided configuration.
 func (ste *simpleTestExecutor) ExecuteTest(ctx Context, conf *api.Config) error {
+	err := ctx.GetFramework().CreateAutomanagedNamespaces(int(conf.AutomanagedNamespaces))
+	defer cleanupResources(ctx)
+	if err != nil {
+		return fmt.Errorf("automanaged namespaces creation failed: %v", err)
+	}
+
+	for i := range conf.Steps {
+		ste.ExecuteStep(ctx, &conf.Steps[i])
+	}
 	return nil
 }
 
 // ExecuteStep executes single test step based on provided step configuration.
 func (ste *simpleTestExecutor) ExecuteStep(ctx Context, step *api.Step) error {
+	if len(step.Measurements) > 0 {
+		// TODO: handle measurments
+	} else {
+		var wg wait.Group
+		defer wg.Wait()
+		for i := range step.Phases {
+			phase := &step.Phases[i]
+			wg.Start(func() {
+				ste.ExecutePhase(ctx, phase)
+			})
+		}
+	}
 	return nil
 }
 
 // ExecutePhase executes single test phase based on provided phase configuration.
 func (ste *simpleTestExecutor) ExecutePhase(ctx Context, phase *api.Phase) error {
+	if phase.NamespaceRange == nil {
+		for j := range phase.ObjectBundle {
+			innerExecuteObject(ctx, &phase.ObjectBundle[j], phase.ReplicasPerNamespace, "")
```

Comment:
You should somehow capture the error returned from `innerExecuteObject1 - else there's no point as you'll always return with nil.

Reply:
In this implementation I though about always passing `nil`. Errors will be logged and not escalated.
On the other hand I need something for returning test states - success or fail.
Let me rethink this.

Reply:
I've added error propagation. I assume that this propagates only errors which are critical (execution should be stopped).